### ISSUE TITLE
 KDE sees my device but not the Firefox extention...

### ISSUE THREAD

**Issue by Jiminimonka** [pdf/kdeconnect-chrome-extension#34]
Manjaro KDE Plasma Kernel 5.6.8

The phone is connected but Firefox KDE icon has a red ! on it.

**Comment by pdf** [comment]
Did you successfully install the native host application per the README? If you click on the extension icon, what does it say?

**Comment by vendion** [comment]
I'm seeing this same issue with Firefox 78.0.2 with FreeBSD 12.1. My phone and computer can see each other, but the Firefox extension just has an Orange exclamation mark and if I click on it there are no devices listed. I did compile the `kdeconnect-chrome-extension` binary and ran `kdeconnect-chrome-extension -install` making sure to select "Firefox" from the list.

![2020-07-22-191509_319x290_scrot](https://user-images.githubusercontent.com/145018/88238529-e5e9aa00-cc4f-11ea-9e91-af0e1bd3fd88.png)

**Comment by pdf** [comment]
This is a different question to the OP @vendion let's continue in #37

**Comment by DefectiveByDesign** [comment]
There is no windows 10 support then?

**Comment by pdf** [comment]
@DefectiveByDesign see #4 #39

**Comment by HarHarLinks** [comment]
I have this problem on arch linux, swaywm, vivaldi-snapshot. kdeconnect-indicator shows my device, vivaldi extension says no devices found. I did ' ./kdeconnect-chrome-extension -install' and selected Vivaldi, assuming this would work for vivaldi-snapshot.

**Comment by pdf** [comment]
@HarHarLinks if vivaldi-snapshot stores its user data in a different location to standard vivaldi, you'll need to copy `~/.config/vivaldi/NativeMessagingHosts/` to the equivalent config directory. I'm afraid I can't test if that's the case though, and I'm not certain how it's packaged.

**Comment by HarHarLinks** [comment]
I did `ln -s ~/.config/vivaldi/NativeMessagingHosts ~/.config/vivaldi-snapshot/NativeMessagingHosts` and it started working. Perhaps it is possible for you to add snapshot to the install menu or create this symlink etc.

---

I am using another NativeMessaging plugin (https://github.com/smorks/keepassnatmsg) that resides in the folder you mention and works from there. The rest of it's config is in `~/.config/vivaldi-snapshot/` but there isn't a second NativeMessagingHosts directory in there which reinforces my assumption that it should have worked in the first place?

**Comment by pdf** [comment]
> I am using another NativeMessaging plugin (https://github.com/smorks/keepassnatmsg) that resides in the folder you mention and works from there.

That's very confusing then. Are you certain you restarted the browser after install?

**Comment by HarHarLinks** [comment]
I had the browser extension already installed from chrome webstore. I then closed the browser before running the binary from the release file. When after starting the browser again it wasn't connected, I submitted my first comment here above.

**Comment by pdf** [comment]
Then I have no idea why one native host would work, and the other wouldn't. Extremely confusing :confused:

**Comment by ChristianNext** [comment]
Connection through Firefox and Chromium extension fails with  
```
Error: no version response received from native host
No devices found
```
This used to work. I can send links from Android phone to my desktop just fine, but not the other way around. Sending files from desktop to Android also works.

Ubuntu 22.04, Firefox and Chromium, kdeconnect-chrome-extension installed

**Comment by pdf** [comment]
@ChristianNext please try re-installing the native host from the releases section of this repository.

**Comment by Megver83** [comment]
Exactly the same issue in Parabola (Arch-like) with Iceweasel (re-branded firefox). KDE Connect is connected to my Android phone but the extension has the exclamation mark. Tried installing the native host system-wide and per-user but nothing.

**Comment by pdf** [comment]
> Exactly the same issue in Parabola (Arch-like) with Iceweasel (re-branded firefox). KDE Connect is connected to my Android phone but the extension has the exclamation mark. Tried installing the native host system-wide and per-user but nothing.

Yiu'll probably need to work out where IceWeasel puts its native hosts and do a custom install.

**Comment by Megver83** [comment]
Iceweasel has the same native hosts paths as firefox

**Comment by pdf** [comment]
> Iceweasel has the same native hosts paths as firefox

I cannot find Iceweasel in any major distributions to test. Even Debian has removed it, and they created it. Unless you mean the GNU version, but that's been Icecat for many years.

**Comment by Megver83** [comment]
Yes, Debian removed it but Parabola continued the branding. Anyways, for some unknown reason now it is working.

**Comment by ReCorno66** [comment]
Having the same problem for some time now, on FireFox 103.0.2 (64-bit) Manjaro Xfce ( kernel Linux 5.15.60-1).
Phone is connected to same local network or connected via USB-port on PC
KDE-Connect client ( version 22.04.3-1 ) on PC is able to connect to the phone

But in FireFox I get this error-message
Error: could not connect to native host
Error: no version response received from native host


Any ideas on this issue?
Thanks, Ronald

**Comment by benyamin-codez** [comment]
I'm pretty sure it's because firefox-snap in jammy jellyfish doesn't support NativeMessaging protocol, per [release notes](url=https://discourse.ubuntu.com/t/jammy-jellyfish-release-notes/24668):

> The Firefox snap does not support the NativeMessaging protocol yet but this feature is planned to be added soon. This means for instance that installing GNOME Shell extensions from Firefox won’t work. As a workaround, you can try the gnome-shell-extension-manager app.

More info:
https://launchpad.net/bugs/1741074
https://bugzilla.mozilla.org/show_bug.cgi?id=1661935
https://forum.snapcraft.io/t/call-for-testing-native-messaging-support-in-the-firefox-snap/31055/31

HTH, Ben

**Comment by kevin-roan** [comment]
Facing same issue on arch with dwm. kde connect notification service works, but browser extension doesn't
- I found that when i open kde-connect-app it shows Device disconnected, i have to wait 2-3 seconds and it connects to the device automatically. and the browser extension works perfectly fine.
- I guess your kde-connect-app should be running on the background.

**Comment by pdf** [comment]
@kevin-roan that behaviour is expected - the browser extension just communicates with the main application, so it has to be running for the extension to work. On KDE for example, it's started automatically by default.

Thanks for noting your experience.

**Comment by snetsplit** [comment]
I had this issue. Long story short it looks like the installer might be crashing when it goes to set permissions.

When ran as root creates the files these files in **/usr/lib/mozilla**

**com.0xc0dedbad.kdeconnect_chrome
com.0xc0dedbad.kdeconnect_chrome.json**

However it fails to set the proper permissions. **com.0xc0dedbad.kdeconnect_chrome** has execute permissions, but nether file is granted read access to nonroot users.

I fixed my permissions and it error went away, and it's working now. Yay!

I did it from dolphin but I think you could use:

```
sudo chmod 755 /lib/mozilla/native-messaging-hosts/com.0xc0dedbad.kdeconnect_chrome
sudo chmod 755 /lib/mozilla/native-messaging-hosts/com.0xc0dedbad.kdeconnect_chrome.json
```
to try to fix it

**Comment by snetsplit** [comment]
Nevermind, now it can list my kde connected devices but the send button is grayed out, and it lists all of them even the offline ones

**Comment by pdf** [comment]
> but nether file is granted read access to nonroot users.

I hadn't considered that users might have a restrictive umask in place, which sounds like the case here, we can set perms explicitly for system-wide install I suppose.


> Nevermind, now it can list my kde connected devices but the send button is grayed out, and it lists all of them even the offline ones

This is #56 I've done most of the work to fix this for recent KDE Connect versions, but probably won't be able to finish it until next week.
